rts/Rendering/Env/Particles/ProjectileDrawer.cpp

```diff
@@ -817,7 +817,9 @@ void CProjectileDrawer::DrawAlpha(bool drawAboveWater, bool drawReflection, bool
 
 		fxShader->Enable();
 
-		fxShader->SetUniform("clipPlane", 0.0f, (drawAboveWater ? 1.0f : -1.0f), 0.0f, 0.0f);
+		const float waterLevel = CGround::GetWaterPlaneLevel();
+
+		fxShader->SetUniform("clipPlane", 0.0f, (drawAboveWater ? waterLevel + 1.0f : waterLevel - 1.0f), 0.0f, 0.0f);
```

Comment:
That's a good point. Do you know where the shader code for this is? 

I'd guess just from conventions that `clipPlane` would be normalize coordinates in -1 to 1, but there's a very slim chance it could also be some other coordinate system that gets translated inside the shader. I don't like chances, especially when I should be able to just go read the shader and know for sure.

Reply:
I found the shader code: https://github.com/beyond-all-reason/spring/blob/a5aa45cc00ea37face1533ad107584d2873fb1a0/cont/base/springcontent/shaders/GLSL/ProjFXVertProg.glsl#L24

I also did some more digging and found a bunch of the code that calls these shaders and sets clip planes at the GL level. It seems like it all assumes -1 is underwater, and +1 is above water.

I'm going to leave this one out of the PR since I think this +-1 values might need to match the ones in other places, but certainly that the shader only really cares about positive or negative distance from the clip place.

I think this is just a trick to clip things above/below the water surface. Since water is current hard coded at y=0, then anything that's underwater will have a negative y for position. If we pass -1 to the shader for the clip plane, then everything that's underwater will end up with a positive distance from the clip plane and be drawn. Above water things (y is positive) will end up with a negative distance and not be drawn. If we pass +1 in as the clip plane, then everything underwater will have a negative distance to plane and not be drawn, etc.

I'm not sure what coordinate system the vertices passed into the shader are using, but if they're using world coordinates then this calculation of the clip plane should definitely use the water level. Otherwise, I think we'd want to translate the water level into whatever coordinate system the shader is using.

I've got a decent idea of how to test this (I just need some units that shoot transparent projectiles), but don't want to dig deep into it just for this. There's a bunch of water shaders I found and I realize the most obvious test would be to see water higher/lower if I set the value to more or less than 0. But without updating the shaders, I think the water won't visually move at all.